# HG changeset patch
# User Daniel J Sebald <daniel.sebald@ieee.org>
# Date 1464802219 18000
#      Wed Jun 01 12:30:19 2016 -0500
# Node ID 6c8c18f49eed4e6ff0b2af5bbad6517b4fd1d539
# Parent  973c16f1c9fce2e06998fd8ff8e72c9d38097cc9
Remove gnuplot graphics view reset after plot for smooth rotation (bug #47974).

* __gnuplot_draw_axes__.m: Only change the view after plotting if gnuplot was
  placed in map view.  Also, rather than 'unset view', set the view to the map
  rotation angles 0,0.

diff --git a/scripts/plot/util/private/__gnuplot_draw_axes__.m b/scripts/plot/util/private/__gnuplot_draw_axes__.m
--- a/scripts/plot/util/private/__gnuplot_draw_axes__.m
+++ b/scripts/plot/util/private/__gnuplot_draw_axes__.m
@@ -1671,19 +1671,19 @@ function __gnuplot_draw_axes__ (h, plot_
                                 have_cdata(i));
       endif
     endfor
     fputs (plot_stream, "\n");
   else
     fputs (plot_stream, "plot \"-\";\nInf Inf\ne\n");
   endif
 
-  ## Needed to allow mouse rotation with pcolor.
-  if (view_map)
-    fputs (plot_stream, "unset view;\n");
+  ## Needed to allow mouse rotation if gnuplot was put in map view.
+  if (view_map && rot_x == 0 && rot_z == 0)
+    fputs (plot_stream, "set view 0,0;\n");
   endif
 
   if (bg_is_set)
     fputs (plot_stream, "unset obj 1;\n");
     bg_is_set = false;
   endif
 
   fflush (plot_stream);
